src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { useUID } from 'react-uid';
+import styles from './ProgressBar.module.css';
+import Text from '../Text';
+
+export type Props = {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Overrides the default "totalValue / max" in cases where a different caption may be desired.
+   */
+  caption?: string;
+  /**
+   * Label associated with naming the progress bar.
+   */
+  label: string;
+  /**
+   * Max value to be represented by the progress bar.
+   */
+  max: number;
+  /**
+   * The amount of segments to be represented by the progress bar.
+   * Recommended max number of segments is 10.
+   */
+  segmentCount: number;
+  /**
+   * The value that each segment represents.
+   */
+  segmentValue: number;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {ProgressBar} from "@chanzuckerberg/eds";`
+ *
+ * A progress bar component that indicates how much progress has been made on a task.
+ */
+export const ProgressBar = ({
+  caption,
+  className,
+  label,
+  segmentCount,
+  segmentValue,
+  max,
+  ...other
+}: Props) => {
+  // Creates an array of length segmentCount and fills them with styled divs to represent the segments.
+  const segments = segmentCount
+    ? Array.from({ length: segmentCount }, (_, index) => {
+        // Calculates the segment width to prevent jumping once progress bar is full
+        const segmentsPerBar = Math.ceil(max / segmentValue);
```

Comment:
How is `segmentsPerBar` different from `segmentCount`?

Reply:
`segmentsPerBar` is the amount of segments in a bar before the bar is "full" so if a `segmentValue` is 2 and the max value is 5, then `segmentsPerBar` would be 3

Reply:
Will rename to `segmentsPerFullBar` to be more intuitive